homeassistant/components/sabnzbd.py

```diff
@@ -0,0 +1,250 @@
+        ssl = info.get('properties', {}).get('https', '0') == '1'
+        async_configure_sabnzbd(hass, info, ssl)
+
+    discovery.async_listen(hass, SERVICE_SABNZBD, sabnzbd_discovered)
+
+    conf = config.get(DOMAIN)
+    if conf is not None:
+        use_ssl = conf.get(CONF_SSL)
+        name = conf.get(CONF_NAME)
+        api_key = conf.get(CONF_API_KEY)
+        await async_configure_sabnzbd(hass, conf, use_ssl, name, api_key)
+    return True
+
+
+async def async_setup_sabnzbd(hass, sab_api, config, name):
+    """Setup SABnzbd sensors and services."""
+    hass.data[DATA_SABNZBD] = SabnzbdApiData(hass, sab_api, name,
+                                             config.get(CONF_SENSORS, {}),
+                                             config.get(CONF_SCAN_INTERVAL))
+    if config.get(CONF_SENSORS):
+        hass.async_add_job(
+            discovery.async_load_platform(hass, 'sensor', DOMAIN, {}, config))
+
+    async def async_service_handler(service):
+        """Handle service calls."""
+        sab_api_data = hass.data[DATA_SABNZBD]
+        if service.service == SERVICE_PAUSE:
+            await sab_api_data.async_pause_queue()
+        elif service.service == SERVICE_RESUME:
+            await sab_api_data.async_resume_queue()
+        elif service.service == SERVICE_SET_SPEED:
+            speed = service.data.get(ATTR_SPEED, None)
+            if speed is not None:
+                await sab_api_data.async_set_queue_speed(speed)
+            else:
+                await sab_api_data.async_set_queue_speed()
+
+    hass.services.async_register(DOMAIN, SERVICE_PAUSE, async_service_handler)
+    hass.services.async_register(DOMAIN, SERVICE_RESUME, async_service_handler)
+    hass.services.async_register(DOMAIN, SERVICE_SET_SPEED,
+                                 async_service_handler,
+                                 schema=SPEED_LIMIT_SCHEMA)
+
+
+def request_configuration(hass, config, host):
+    """Request configuration steps from the user."""
+    from pysabnzbd import SabnzbdApi
+
+    configurator = hass.components.configurator
+    # We got an error if this method is called while we are configuring
+    if host in _CONFIGURING:
+        configurator.notify_errors(_CONFIGURING[host],
+                                   'Failed to register, please try again.')
+
+        return
+
+    async def async_configuration_callback(data):
+        """Handle configuration changes."""
+        api_key = data.get(CONF_API_KEY)
+        sab_api = SabnzbdApi(host, api_key)
+        if await async_check_sabnzbd(sab_api):
+
+            def success():
+                """Setup was successful."""
+                conf = load_json(hass.config.path(CONFIG_FILE))
+                conf[host] = {CONF_API_KEY: api_key}
+                save_json(hass.config.path(CONFIG_FILE), conf)
+                req_config = _CONFIGURING.pop(host)
+                configurator.async_request_done(req_config)
+
+            success()
```

Comment:
You removed the call to `hass.async_add_job` from the previous code, and so you're doing I/O inside the event loop 😞